src/resolvelib/resolvers.py

```diff
@@ -369,11 +371,12 @@ def resolve(self, requirements, max_rounds):
                 # Backtrack if pinning fails. The backtrack process puts us in
                 # an unpinned state, so we can work on it in the next round.
                 success = self._backtrack()
+                self._failure_causes = [i for c in failure_causes for i in c.information]
 
                 # Dead ends everywhere. Give up.
                 if not success:
                     causes = [i for c in failure_causes for i in c.information]
```

Comment:
Can't we drop the (re-) calculation of causes here?

Reply:
Fixed. I think this error and the typo crept in when I had to split up my original patch between the different repositories, apologies for that.